### PR TITLE
Bugfix: Fix issue connecting to redis when specifying password file

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
Trim whitespace from password file, similar to how [prometheus does it](https://github.com/prometheus/common/blob/8c9cb3fa6d01832ea16937b20ea561eed81abd2f/config/http_config.go#L625-L629).

#### Which issue(s) this PR fixes
[#2435](https://github.com/grafana/agent/issues/2435), at least partially. This will fix whitespace being sent as part of password when specifying a password file but does not fix the issue of the environment variable not being used.

#### PR Checklist

- [X] CHANGELOG updated
- [ ] Documentation added
- [X] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -111,6 +111,8 @@ Main (unreleased)
 
 - Support Bearer tokens for metric remote writes in the Grafana Operator (@jcreixell, @marctc)
 
+- Fix issue when using a password file for `redis_exporter`. (@spartan0x117)
```
Comment (rfratto) [inline CHANGELOG.md:114]: Can you update the changelog entry [to describe what the issue was?](https://github.com/grafana/agent/blob/main/docs/developer/contributing.md#changelog-entry-style-tips)

**Comment by rfratto** [comment]
This feels like a breaking change to me, since it means a file containing just the password won't work anymore. Do we understand why the existing code isn't working?

**Comment by spartan0x117** [comment]
Yeah, as-is I realized this would be a breaking change so added a TODO to fallback to the old way (with the slight modification of trimming any newlines at the end, if they exist).

The original issue looks to have been caused when a newline was placed at the end of the file. The newline would be read as a part of the `os.ReadFile` and then would be passed to redis for authentication and most likely fail.

We could solve this either by doing the `re.LoadPwdFile`, falling back to `os.ReadFile` + trimming newlines, _or_ just `os.ReadFile` + trimming newlines. I'm happy to do either way

**Comment by rfratto** [comment]
I'd prefer `os.ReadFile` + trimming newlines, since Prometheus will [also trim newlines when reading password files](https://github.com/prometheus/common/blob/8c9cb3fa6d01832ea16937b20ea561eed81abd2f/config/http_config.go#L629). 

If we want to support the password map from redis_exporter, personally I'd suggest making it something you have to opt into by setting a flag or through a completely distinct `password_map_file` setting. I don't think this is necessary for fixing #2435, but you're more than free to add it if you think supporting the password map file is important :)

File: CHANGELOG.md (hunk 1 above)
Comment (spartan0x117) [reply CHANGELOG.md]: Done!

**Review by rfratto — APPROVED** [review]
LGTM, thanks! 

I'd like to hold off on merging until Monday after the v0.29.0-rc.0 CHANGELOG entries are brought over to main to avoid any confusion with which changes go where.

**Pull request merged**